<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Process and Analyze Nanopore Sequencing Methylation Data • ModSeqR</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Process and Analyze Nanopore Sequencing Methylation Data">
<meta name="description" content="ModSeqR creates methylation databases and processes them with customizable options. Data can be summarized by positions, windows, or provided genomic annotations (e.g., BED files). The package includes quality control, differential methylation analysis, and sliding window support.">
<meta property="og:description" content="ModSeqR creates methylation databases and processes them with customizable options. Data can be summarized by positions, windows, or provided genomic annotations (e.g., BED files). The package includes quality control, differential methylation analysis, and sliding window support.">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">ModSeqR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="articles/index.html">Articles</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/&lt;your-github-username&gt;/ModSeqR"><span class="fa fab fa-github"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><p><img src="reference/figures/WBL_MODSEQR.png" style="width: 275px;"></p>
<div class="section level1">
<div class="page-header"><h1 id="modseqr">ModSeqR<a class="anchor" aria-label="anchor" href="#modseqr"></a>
</h1></div>
<div class="section level2">
<h2 id="version-100">Version 1.0.0<a class="anchor" aria-label="anchor" href="#version-100"></a>
</h2>
<p><strong>(Updated December 29th 2025)</strong></p>
<p><strong><em>Note</em></strong>: <em>This is an early release - changes may occur that significantly change the functionality and structure of the data and functions. The user should be aware that subsequent releases may break code written using earlier releases.</em></p>
<p><strong>ModSeqR</strong> is an R package managing Direct Whole Methylome Sequencing (dWMS) data. It creates a database, and processes it with unique options. Data can be summarized by positions, windows, or provided an annotation bed file, by unique genomic regions. The package also offers quality control functions, differential methylation, and a sliding window analysis.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>For easy visualization and data management, we encourage you to download and use RStudio. RStudio download instructions can be found <a href="https://posit.co/download/rstudio-desktop/" class="external-link">here</a>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Install the devtools package if necessary</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Install ModSeqR from GitHub</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"Wasatch-Biolabs-Bfx/ModSeqR"</span>, build_vignettes <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Access Package</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/Wasatch-Biolabs-Bfx/ModSeqR" class="external-link">ModSeqR</a></span><span class="op">)</span></span></code></pre></div>
<p><strong><em>For Linux Users:</em></strong> <em>System packages may need to be intalled in order to use devtools. Instructions can be found online. An example guide for this can be found <a href="https://www.digitalocean.com/community/tutorials/how-to-install-r-packages-using-devtools-on-ubuntu-16-04" class="external-link">here</a>.</em></p>
</div>
<div class="section level2">
<h2 id="creating-ch3-files">Creating .ch3 Files<a class="anchor" aria-label="anchor" href="#creating-ch3-files"></a>
</h2>
<p>If your sequencing data is coming from <a href="https://www.wasatchbiolabs.com/" class="external-link">Wasatch Biolabs</a>, <code>.ch3</code> files can be delivered directly with your batch upon request. These files contain methylation calls from a third-generation sequencing run. Typically, one .ch3 file is created per sample, and each file compresses large amounts of raw data into a usable intermediate format.</p>
<p>If .ch3 files were not provided, you can build them yourself using the make_ch3_archive() function in this package. This takes a tab-delimited file of methylation calls (such as the output of modkit extract-calls) and compresses it into .ch3 format for downstream use.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Convert a calls.tsv file to compressed .ch3 format</span></span>
<span><span class="fu"><a href="reference/make_mod_archive.html">make_mod_archive</a></span><span class="op">(</span></span>
<span>  file_name   <span class="op">=</span> <span class="st">"calls.tsv"</span>,   <span class="co"># input modkit calls file</span></span>
<span>  sample_name <span class="op">=</span> <span class="st">"sample1"</span>,     <span class="co"># will be embedded in output filenames</span></span>
<span>  out_path    <span class="op">=</span> <span class="st">"output_dir/"</span>, <span class="co"># where to write .ch3 files</span></span>
<span>  short_ids   <span class="op">=</span> <span class="cn">TRUE</span>           <span class="co"># optionally shorten read_id to reduce size</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="important">Important:<a class="anchor" aria-label="anchor" href="#important"></a>
</h3>
<ul>
<li>.ch3 files use 0-based, half-open genome coordinates (like BED files).</li>
<li>Example: a CpG at base 1000 (1-based) will appear as start=999, end=1001.</li>
<li>Each output archive is written in compressed Parquet format (zstd), typically producing multiple .ch3 files per sample (e.g., sample1-0.ch3, sample1-1.ch3).</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="example-data">Example Data<a class="anchor" aria-label="anchor" href="#example-data"></a>
</h2>
<p>If you’d like to test the package without generating your own .ch3 files, small example data are included with the package in:</p>
<p><code>ModSeqR/inst/extdata/ch3_files/</code></p>
<p>You can download these example .ch3 files directly from the GitHub repository under the inst/extdata/ch3_files folder.</p>
<p>These test files can be used to practice building a database and running the full analysis workflow.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Example: build a test database using included example files</span></span>
<span><span class="va">example_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/ch3_files"</span>, package <span class="op">=</span> <span class="st">"ModSeqR"</span><span class="op">)</span></span>
<span><span class="va">mod_db</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/make_mod_db.html">make_mod_db</a></span><span class="op">(</span><span class="va">example_path</span>, db_name <span class="op">=</span> <span class="st">"example_db"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="instructions">Instructions<a class="anchor" aria-label="anchor" href="#instructions"></a>
</h2>
<p>Begin with .ch3 files created by Wasatch Biolabs or individually, and build a database using <code><a href="reference/make_mod_db.html">make_mod_db()</a></code>. This will at first hold a calls table. If you would like to see key stats on your CH3 file, call <code><a href="reference/get_mod_stats.html">get_mod_stats()</a></code>. This shows information like CpG coverage, calls by flag value, high confidence calls, high quality calls, and average read length.</p>
<div class="section level3">
<h3 id="summarize-data">Summarize Data<a class="anchor" aria-label="anchor" href="#summarize-data"></a>
</h3>
<p>After a database is created, a user can summarize their data by position (<code><a href="reference/summarize_mod_positions.html">summarize_mod_positions()</a></code>), by regions (<code><a href="reference/summarize_mod_regions.html">summarize_mod_regions()</a></code>), by windows (<code><a href="reference/summarize_mod_windows.html">summarize_mod_windows()</a></code>), or by reads ((<code><a href="reference/summarize_mod_reads.html">summarize_mod_reads()</a></code>).</p>
</div>
<div class="section level3">
<h3 id="differential-methylation">Differential Methylation<a class="anchor" aria-label="anchor" href="#differential-methylation"></a>
</h3>
<p>A differential methylation analysis can be conducted on positional, regional, or window data using <code><a href="reference/calc_mod_diff.html">calc_mod_diff()</a></code>. After calculating methylation differences between windows, use <code><a href="reference/collapse_mod_windows.html">collapse_mod_windows()</a></code> to collapse significant windows in a methylation dataset. This merges contiguous regions that meet the specified criteria.</p>
</div>
<div class="section level3">
<h3 id="get-database-stats">Get Database Stats<a class="anchor" aria-label="anchor" href="#get-database-stats"></a>
</h3>
<p>If you would like to see key stats on your database at any time, including what unique sample names are in the data for a differential analysis, call <code><a href="reference/get_mod_dbinfo.html">get_mod_dbinfo()</a></code>. To see what columns are in a table in your database and how many records (rows) there are, call <code><a href="reference/get_mod_tableinfo.html">get_mod_tableinfo()</a></code> with your database and desired table name.</p>
</div>
<div class="section level3">
<h3 id="quality-control">Quality Control<a class="anchor" aria-label="anchor" href="#quality-control"></a>
</h3>
<p><code><a href="reference/run_mod_qc.html">run_mod_qc()</a></code> can be called to visually assess any data. Running a QC can take a long time on large data, so set the argument <code>max_rows</code> to a reasonable value (ex. 1000) to assess data faster. To view and extract a table, call <code>export_table()</code> to export any data table from the database to a file, or use <code>get_table()</code> to import as a tibble into your local environment. Similarily, use <code>max_calls</code> if you are fine with a smaller, randomized set of data.</p>
<p>If you would like to run everything in one command, call <code><a href="reference/run_mod_analysis.html">run_mod_analysis()</a></code>.</p>
</div>
</div>
<div class="section level2">
<h2 id="paradigm">Paradigm<a class="anchor" aria-label="anchor" href="#paradigm"></a>
</h2>
<p>You can pipe your functions together, or feel free to call each function one line at a time. Below are two examples of this.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/getwd.html" class="external-link">setwd</a></span><span class="op">(</span><span class="st">"/home/directory/analysis"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Build database and run analysis in a pipe</span></span>
<span><span class="va">mod_db</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/make_mod_db.html">make_mod_db</a></span><span class="op">(</span></span>
<span>  ch3_files <span class="op">=</span> <span class="st">"../ch3_files_directory"</span>, </span>
<span>             db_name <span class="op">=</span> <span class="st">"my_data"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="reference/summarize_mod_windows.html">summarize_mod_windows</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="reference/calc_mod_diff.html">calc_mod_diff</a></span><span class="op">(</span>call_type <span class="op">=</span> <span class="st">"windows"</span>,</span>
<span>              cases <span class="op">=</span></span>
<span>                <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"sperm"</span><span class="op">)</span>,</span>
<span>              controls <span class="op">=</span></span>
<span>                <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"blood"</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="reference/collapse_mod_windows.html">collapse_mod_windows</a></span><span class="op">(</span><span class="op">)</span> </span>
<span>  </span>
<span><span class="co"># Build and analyze through separate lines</span></span>
<span><span class="va">mod_db</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/make_mod_db.html">make_mod_db</a></span><span class="op">(</span>ch3_files <span class="op">=</span> <span class="st">"../ch3_files_directory"</span>, db_name <span class="op">=</span> <span class="st">"my_data"</span><span class="op">)</span></span>
<span><span class="va">mod_db</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/summarize_mod_windows.html">summarize_mod_windows</a></span><span class="op">(</span><span class="va">mod_db</span><span class="op">)</span></span>
<span><span class="va">mod_db</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/calc_mod_diff.html">calc_mod_diff</a></span><span class="op">(</span><span class="va">mod_db</span>, call_type <span class="op">=</span> <span class="st">"windows"</span>, cases <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"sperm"</span><span class="op">)</span>, controls <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"blood"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">mod_db</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/collapse_mod_windows.html">collapse_mod_windows</a></span><span class="op">(</span><span class="va">mod_db</span><span class="op">)</span> </span>
<span></span>
<span><span class="co"># Run entire differential methylation analysis in one function</span></span>
<span><span class="fu"><a href="reference/run_mod_analysis.html">run_mod_analysis</a></span><span class="op">(</span><span class="va">mod_db</span>, </span>
<span>             out_path <span class="op">=</span> <span class="st">"/analysis"</span>,</span>
<span>             call_type <span class="op">=</span> <span class="st">"windows"</span>,</span>
<span>             cases <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"sperm"</span><span class="op">)</span>,</span>
<span>             controls <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"blood"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Check to see what's in your database at any time</span></span>
<span><span class="fu"><a href="reference/get_mod_dbinfo.html">get_mod_dbinfo</a></span><span class="op">(</span><span class="va">mod_db</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Check to see the columns in any table </span></span>
<span><span class="fu"><a href="reference/get_mod_tableinfo.html">get_mod_tableinfo</a></span><span class="op">(</span><span class="va">mod_db</span>, <span class="st">"windows"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Export differentially methylated data to your computer</span></span>
<span><span class="fu"><a href="reference/export_mod_table.html">export_mod_table</a></span><span class="op">(</span><span class="va">mod_db</span>, <span class="st">"collapsed_windows"</span>, <span class="st">"../results_directory"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># OR, to work with your data locally in your R environment</span></span>
<span><span class="va">positions</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/get_mod_table.html">get_mod_table</a></span><span class="op">(</span><span class="va">mod_db</span>, <span class="st">"positions"</span><span class="op">)</span></span>
<span><span class="va">windows</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/get_mod_table.html">get_mod_table</a></span><span class="op">(</span><span class="va">mod_db</span>, <span class="st">"windows"</span><span class="op">)</span></span>
<span><span class="va">regions</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/get_mod_table.html">get_mod_table</a></span><span class="op">(</span><span class="va">mod_db</span>, <span class="st">"regions"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#DONE! Data has been analyzed and exported!</span></span></code></pre></div>
<p><strong><em>Warning</em></strong> <em>- If using samples other than human or with unique chromosome names, remember to adjust the chrs argument in each function!</em></p>
<div class="section level3">
<h3 id="convenience-functions">Convenience Functions<a class="anchor" aria-label="anchor" href="#convenience-functions"></a>
</h3>
<p>ModSeqR also provides a few helper utilities to make it easier to inspect and manage your database:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># View all column names in a given table</span></span>
<span><span class="fu"><a href="reference/get_mod_cols.html">get_mod_cols</a></span><span class="op">(</span><span class="va">mod_db</span>, <span class="st">"calls"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Count unique CpG sites (based on start/end)</span></span>
<span><span class="fu"><a href="reference/get_mod_cpg_count.html">get_mod_cpg_count</a></span><span class="op">(</span><span class="va">mod_db</span>, table_name <span class="op">=</span> <span class="st">"calls"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Show high-level database statistics (size, tables, sample names)</span></span>
<span><span class="fu"><a href="reference/get_mod_dbinfo.html">get_mod_dbinfo</a></span><span class="op">(</span><span class="va">mod_db</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get detailed information about a specific table</span></span>
<span><span class="fu"><a href="reference/get_mod_tableinfo.html">get_mod_tableinfo</a></span><span class="op">(</span><span class="va">mod_db</span>, <span class="st">"positions"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Rename sample names inside any table</span></span>
<span><span class="fu"><a href="reference/rename_mod_samples.html">rename_mod_samples</a></span><span class="op">(</span><span class="va">mod_db</span>, <span class="st">"positions"</span>,</span>
<span>                   samples_map <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"old_name"</span> <span class="op">=</span> <span class="st">"new_name"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Remove a table from the database</span></span>
<span><span class="fu"><a href="reference/remove_mod_table.html">remove_mod_table</a></span><span class="op">(</span><span class="va">mod_db</span>, <span class="st">"temp_table"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="getting-help">Getting Help<a class="anchor" aria-label="anchor" href="#getting-help"></a>
</h2>
<p>To see detailed documentation on a specific function in R, call <code>?{function}</code>. Example:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">?</span><span class="fu"><a href="reference/make_mod_db.html">make_mod_db</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>This will render development documentation for that function in the Help tab in Rstudio</p>
<div class="section level3">
<h3 id="vignette">Vignette<a class="anchor" aria-label="anchor" href="#vignette"></a>
</h3>
<p>To get detailed instructions and help working through the package, download and view our vignette in this github repo at <code>docs/ModSeqRWalkthrough.html</code>.</p>
<p>Or, build and follow along the vignette by calling:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/browseVignettes.html" class="external-link">browseVignettes</a></span><span class="op">(</span><span class="st">"ModSeqR"</span><span class="op">)</span></span></code></pre></div>
<p>in R and click on HTML in your browser. Or, to browse the vignette in your R environment, call</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/vignette.html" class="external-link">vignette</a></span><span class="op">(</span><span class="st">"ModSeqRWalkthrough"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="license">License<a class="anchor" aria-label="anchor" href="#license"></a>
</h2>
<p>This package is distributed under the <strong>Personal and Internal Research License (v1.1)</strong>, based on the <a href="https://polyformproject.org/licenses/noncommercial/1.0.0" class="external-link">PolyForm Noncommercial License 1.0.0</a>, with an additional grant of rights for internal research use provided by Wasatch Biolabs.</p>
<ul>
<li>✅ Free for personal use, academic research, educational, and other noncommercial purposes.<br>
</li>
<li>✅ Commercial organizations may also use it internally for research and evaluation.<br>
</li>
<li>❌ Not permitted for commercial services, fee-for-service analysis, redistribution, or production use.</li>
</ul>
<p>See the full license text in the <a href="./LICENSE">LICENSE</a> file included with this repository.</p>
<p>If you have any suggestions or requested features, please email Jonathon Hill at <a href="mailto:jhill@byu.edu" class="email">jhill@byu.edu</a>.</p>
<div class="section level4">
<h4 id="developed-by-wasatch-biolabs">Developed by Wasatch Biolabs.<a class="anchor" aria-label="anchor" href="#developed-by-wasatch-biolabs"></a>
</h4>
</div>
<div class="section level4">
<h4 id="visit-us-on-our-website-for-more-details">Visit us on <a href="https://www.wasatchbiolabs.com/" class="external-link">our website</a> for more details.<a class="anchor" aria-label="anchor" href="#visit-us-on-our-website-for-more-details"></a>
</h4>
<p><img src="reference/figures/wbl_main_logo.png" style="width: 275px;"></p>
</div>
</div>
</div>

  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/Wasatch-Biolabs-Bfx/ModSeqR/" class="external-link">Browse source code</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing ModSeqR</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Jonathon Hill <br><small class="roles"> Author, maintainer </small>   </li>
<li>Hailey Zimmerman <br><small class="roles"> Author </small>   </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Jonathon Hill, Hailey Zimmerman.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
